packages/x-components/src/wiring/__tests__/debounce-operator.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { createResultStub } from '../../__stubs__/index';
+import { debounce } from '../wires.operators';
+import { createWire } from './operators-testing-utils';
+
+describe(`testing ${debounce.name} operator`, () => {
+  beforeAll(jest.useFakeTimers);
+  afterAll(jest.useRealTimers);
+
+  it('discards emitted values that take less than the specified time between output', () => {
+    const { wire, callback, emitWireEvent, registerWire } = createWire();
+    registerWire(debounce(wire, 500));
+    emitWireEvent('1', '2', '3');
+
+    expect(callback).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(499);
+    expect(callback).not.toHaveBeenCalled();
+
+    emitWireEvent('3', '4');
+    jest.advanceTimersByTime(499);
+    expect(callback).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(1);
+    expect(callback).toHaveBeenCalledTimes(1);
+    expect(callback).toHaveBeenCalledWith('4');
```

Comment:
I'm not using `toHaveBeenNthCalledWith` on purpose. As I'm testing only one call at a time, I'm doing so immediately after it happens, and each parameter has a different value, I can simplify and not specify the `nth` of each call.